XUtils.h

```diff
@@ -3,13 +3,21 @@
 /*
 htop - StringUtils.h
 (C) 2004-2011 Hisham H. Muhammad
+(C) 2020-2023 htop dev team
 Released under the GNU GPLv2+, see the COPYING file
 in the source distribution for its full text.
 */
 
+#ifndef PACKAGE
+// strchrnul() needs _GNU_SOURCE defined, see PR #1337 for details
+#error "Must have #include \"config.h\" line at the top of the file that includes these XUtils helper functions"
+#endif
+
+// Workaround to make IWYU happy regarding the used symbols below; translates to noop.
```

Comment:
I'm curious. What error would IWYU generate if we don't have this workaround and just have the `#include "config.h"` line removed?

And I wonder if this line works:
`// IWYU pragma: no_include "config.h"`

Reply:
For `XUtils.c`/`XUtils.h` it would generate "must include `config.h`" message due to the various defines we use from that file in the code below. Strictly speaking that include is covered by the `IWYU pragma: keep` in `XUtils.c`.

FWIW, I mostly added that explicit include for overall consistency; i.e. having the headers actually have an include for things they rely on. As noted in the comment: By the point that include is made it should generate a no-op, or the pre-processor is borked …

Reply:
Used your suggested `IWYU pragma: no_include "config.h"` as a marker.